apps/files/js/filelist.js

```diff
@@ -1787,8 +1787,11 @@
 			td.append(linkElem);
 			tr.append(td);
 
-			var enabledThemes = window.OCA?.Theming?.enabledThemes || []
-			var isDarkTheme = enabledThemes.join('').indexOf('dark') !== -1
+			const enabledThemes = window.OCA?.Theming?.enabledThemes || []
+			// Check enabled themes, if system default is selected check the browser
+			const isDarkTheme = enabledThemes && (enabledThemes.length > 1 || enabledThemes[0] != 'default')
+				? enabledThemes.join('').indexOf('dark') !== -1
+				: window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches
```

Comment:
```suggestion
			const isDarkTheme = (!enabledThemes.length || enabledThemes[0] === 'default')
				? window.matchMedia('(prefers-color-scheme: dark)').matches
				: enabledThemes.join('').indexOf('dark') !== -1
```
Let's invert the logic

The `&&`'s don't seem necessary?

Reply:
You are right, fixed that.